## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#20406ea4-0408-419e-83e9-6c12950f69ef) |
| mint-compression-dist-xl.sh | :x: [more...](#0ff24bb3-3bdb-46f1-b14d-7b0f389dbcfa) |
| mint-dist-xl.sh | :x: [more...](#e95f2bfa-5545-4e34-bc9d-3a03c1f6c778) |
<a name="e95f2bfa-5545-4e34-bc9d-3a03c1f6c778"></a>
### 7431-3d346f9/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.61:31898
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp f857a90e38c0:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-0ky9qfjnzq4fyx",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 363,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.61:31898/aws-sdk-go-test-0ky9qfjnzq4fyx: dial tcp 72.28.97.61:31898: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="0ff24bb3-3bdb-46f1-b14d-7b0f389dbcfa"></a>
### 7431-3d346f9/mint-compression-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.59:31274
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 3362a84b7d8c:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-emae3lrm4nvgrb",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 333,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.59:31274/aws-sdk-go-test-emae3lrm4nvgrb: dial tcp 72.28.97.59:31274: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="20406ea4-0408-419e-83e9-6c12950f69ef"></a>
### 7431-3d346f9/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.53:32556
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 37cc22ad6953:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 2 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 1 minutes and 6 seconds
(4/14) Running aws-sdk-ruby tests ... done in 37 seconds
(5/14) Running awscli tests ... done in 2 minutes and 32 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 26 seconds
(8/14) Running minio-dotnet tests ... done in 2 minutes and 28 seconds
(9/14) Running minio-go tests ... done in 5 minutes and 10 seconds
(10/14) Running minio-java tests ... done in 15 minutes and 59 seconds
(11/14) Running minio-js tests ... done in 2 minutes and 10 seconds
(12/14) Running minio-py tests ... FAILED in 9 minutes and 25 seconds
{
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)",
  "name": "minio-py:test_presigned_get_object_response_headers",
  "status": "FAIL",
  "duration": 4284,
  "args": {
    "bucket_name": "minio-py-test-f3d8a808-8877-4d3e-b920-1f40355b64c8",
    "object_name": "067f6aa9-50df-497d-ab00-77af937cabd6"
  },
  "message": "'content-language'",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1266, in test_presigned_get_object_response_headers\n    returned_content_language = response.headers['Content-Language']\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/_collections.py\", line 152, in __getitem__\n    val = self._container[key.lower()]\nKeyError: 'content-language'\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1888, in main\n    test_presigned_get_object_response_headers(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1274, in test_presigned_get_object_response_headers\n    raise Exception(err)\nException: 'content-language'\n"
}

Executed 11 out of 14 tests successfully.
```

--- a/cmd/http/server.go
+++ b/cmd/http/server.go
@@ -57,6 +57,8 @@ const (
 type Server struct {
 	http.Server
 	Addrs                  []string      // addresses on which the server listens for new connection.
+	ReadTimeout            time.Duration // timeout used for net.Conn.Read() deadlines.
+	WriteTimeout           time.Duration // timeout used for net.Conn.Write() deadlines.
 	ShutdownTimeout        time.Duration // timeout used for graceful server shutdown.
 	TCPKeepAliveTimeout    time.Duration // timeout used for underneath TCP connection.
 	UpdateBytesReadFunc    func(int)     // function to be called to update bytes read in bufConn.

